### PR TITLE
Feature/livedata implementation

### PR DESCRIPTION
**Pull request by Srishtikumari9**

### CODE REVIEW

File: app/src/main/java/com/example/socialapp/network/RetrofitClient.java [hunk 1]
```diff
@@ -15,14 +18,16 @@ private RetrofitClient() {
         interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
         OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
 
-        Retrofit retrofitAuth = new Retrofit.Builder().baseUrl("https://graph.facebook.com/v13.0/")
+        Retrofit retrofitAuth = new Retrofit.Builder()
+                .baseUrl("https://graph.facebook.com/v13.0/")
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/network/RetrofitClient.java]: Please move this hardcoded url to constant.

File: app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java [hunk 2]
```diff
@@ -44,23 +39,17 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                List<Post> posts = response.body().getPosts();
-
+        postViewModel.getPosts("name,description", accessToken).observe(this, response -> {
+            if (response.isSuccessful()) {
+                List<Post> posts = response.body.getPosts();
                 for (Post post : posts) {
                     Log.i(TAG, "" + post.getName());
                 }
+            } else {
+                Log.d(TAG, "PostsError = ");
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java]: Please append the error message.

**Review by pawankgupta-se — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/socialapp/network/RetrofitClient.java (hunk 1 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/network/RetrofitClient.java]: Comments Implemented.

File: app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java (hunk 2 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java]: Comments Implemented.

**Review by pawankgupta-se — APPROVED** [review]

**Pull request merged**